### PR TITLE
Add event processing details to eventing mesh documentation

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Add a new `details` document on event processing
- Include a description of Trigger CR

**Related issue(s)**
See also #7498

### CODE REVIEW

File: docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md [hunk 1]
```diff
@@ -18,32 +17,16 @@ This diagram shows how the Eventing Mesh components work together.
...
 
 4. The Application Broker watches the creation of the Application CR and performs the following actions:
 
     * Exposes the Events API of an external system as a ServiceClass. Once the user provisions this ServiceClass in the Namespace, the Application Broker makes events available to use.
 
     * Deploys Knative Subscription and defines the Broker as the subscriber for the Channel to allow communication between them.
 
-    * Adds the `knative-eventing-injection` label to the user's Namespace. As a result, the Namespace controller creates the [Broker](https://knative.dev/docs/eventing/broker-trigger/) which acts as an entry point for the events. 
+    * Adds the `knative-eventing-injection` label to the user's Namespace. As a result, the Namespace controller creates the [Broker](https://knative.dev/docs/eventing/broker-trigger/) which automatically receives the name `default`. The Broker acts as an entry point for the events and them at the cluster-local endpoint it exposes.
```
Comment (kazydek) [inline docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md:28]:
```suggestion
    * Adds the `knative-eventing-injection` label to the user's Namespace. As a result, the Namespace controller creates the [Broker](https://knative.dev/docs/eventing/broker-trigger/) which automatically receives the `default` name. The Broker acts as an entry point for the events and receives them at the cluster-local endpoint it exposes.
```

File: docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md [hunk 2]
```diff
@@ -18,32 +17,16 @@ This diagram shows how the Eventing Mesh components work together.
...
-
-2. The Subscription defines the Broker as the subscriber. This way, the Channel can communicate with the Broker to send events.
-
-3. The Channel listens for incoming events. When it receives an event, the underlying messaging layer dispatches it to the Broker.
-
-4. The Broker sends the event to the Trigger which is configured to receive events of this type. 
-
-5. The Trigger checks if the attributes of the incoming event match its specification. If they do, the Trigger sends the event to a subscriber, such as a lambda.
\ No newline at end of file
+For details on the Trigger specification, [this](/components/knative-eventing-mesh/#details-event-processing-and-delivery) document.
```
Comment (kazydek) [inline docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md:32]:
```suggestion
For details on the Trigger specification, read [this](/components/knative-eventing-mesh/#details-event-processing-and-delivery) document.
```

File: docs/knative-eventing-mesh/03-01-event-processing.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+---
+title: Event processing and delivery
+type: Details
+---
+The event processing and delivery flow in Knative Eventing Mesh uses the Broker and Trigger concept to forward events and deliver them to the subscribers.
+This diagram explains the event flow in Kyma, from the moment the Application sends an event, to the point when the event triggers the function.
```
Comment (kazydek) [inline docs/knative-eventing-mesh/03-01-event-processing.md:6]:
```suggestion
This diagram explains the event flow in Kyma, from the moment the Application sends an event, to the point when the event triggers the lambda.
```

Comment (kazydek) [reply docs/knative-eventing-mesh/03-01-event-processing.md:6]: or lambda function?

File: docs/knative-eventing-mesh/03-01-event-processing.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+![Eventing flow](./assets/eventing-mesh-flow.svg)
+
```
Comment (kazydek) [inline docs/knative-eventing-mesh/03-01-event-processing.md:9]: could you write on the diagram Function CR (Lambda)?

File: docs/knative-eventing-mesh/03-01-event-processing.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+---
+title: Event processing and delivery
+type: Details
+---
+The event processing and delivery flow in Knative Eventing Mesh uses the Broker and Trigger concept to forward events and deliver them to the subscribers.
```
Comment (kazydek) [inline docs/knative-eventing-mesh/03-01-event-processing.md:5]:
```suggestion
The event processing and delivery flow in Knative Eventing Mesh uses the Broker and Trigger concepts to forward events and deliver them to the subscribers.
```

File: docs/knative-eventing-mesh/03-01-event-processing.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      name: test-lambda # Lambda name
+```
+
+In Kyma, the filter specification defines the Broker which receives events and parameters you must provide for the Trigger to forward events to subscribers. The table lists the mandatory parameters along with their descriptions:
+
+| Parameter  |  Description  |
+|------------|-------------- |
+| **spec.broker** | Name of the Broker that receives events. By default, it receives the value `default` when the user Namespace is labeled with `knative-eventing-injection`.  |
+| **spec.filter.attributes.type** | Specific event type to which you want to subscribe your lambda, such as `user.created`.|
+| **spec.filter.attributes.eventtypeversion** | Event version, such as `v1`. |
```
Comment (kazydek) [inline docs/knative-eventing-mesh/03-01-event-processing.md:48]:
```suggestion
| **spec.filter.attributes.eventTypeVersion** | Event version, such as `v1`. |
```

File: docs/knative-eventing-mesh/03-01-event-processing.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: test-lambda # Lambda name
+```
+
+In Kyma, the filter specification defines the Broker which receives events and parameters you must provide for the Trigger to forward events to subscribers. The table lists the mandatory parameters along with their descriptions:
+
+| Parameter  |  Description  |
+|------------|-------------- |
+| **spec.broker** | Name of the Broker that receives events. By default, it receives the value `default` when the user Namespace is labeled with `knative-eventing-injection`.  |
```
Comment (kazydek) [inline docs/knative-eventing-mesh/03-01-event-processing.md:46]:
```suggestion
| **spec.broker** | Name of the Broker that receives events. By default, it receives the value `default` when the user's Namespace is labeled with `knative-eventing-injection`.  |
```

File: docs/knative-eventing-mesh/03-01-event-processing.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      eventtypeversion: v1 # Event version
+      source: mock # Application name
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: test-lambda # Lambda name
+```
+
+In Kyma, the filter specification defines the Broker which receives events and parameters you must provide for the Trigger to forward events to subscribers. The table lists the mandatory parameters along with their descriptions:
```
Comment (kazydek) [inline docs/knative-eventing-mesh/03-01-event-processing.md:42]: `the filter specification`?

Comment (kazydek) [reply docs/knative-eventing-mesh/03-01-event-processing.md:42]: and maybe sth like: `The **filter** section of the Trigger CR specification defines the Broker which receives events and parameters you must provide for the Trigger to forward events to subscribers. These parameters are not fixed and depend on a given use case. In Kyma, these are the mandatory ones: `

File: docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md (hunk 1 above)
Comment (bszwarc) [reply docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md]: I changed it as agreed :)

File: docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md [hunk 9]
```diff
@@ -18,32 +17,16 @@ This diagram shows how the Eventing Mesh components work together.
 
     * [HTTP Source Adapter](https://github.com/kyma-project/kyma/tree/master/components/event-sources/adapter/http) which is an HTTP server deployed inside the `kyma-integration` Namespace. This adapter acts as a gateway to the Channel, and is responsible for exposing an endpoint to which the Application sends the events. 
 
-    * [Channel](https://knative.dev/docs/eventing/channels/) which defines the way messages are dispatched in the Namespace. Its underlying implementation is responsible for forwarding events to the Broker or additional Channels. Kyma uses NATS Streaming as its default Channel, but you can change it to InMemoryChannel, Kafka, or Google PubSub. 
+    * [Channel](https://knative.dev/docs/eventing/channels/) which defines the way messages are dispatched in the Namespace. Its underlying implementation is responsible for forwarding events to the Broker or additional Channels. Kyma uses NATS Streaming as its default Channel, but you can change it to Kafka implementation.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md]: Added it back as agreed :)

File: docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md (hunk 2 above)
Comment (bszwarc) [reply docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md]: 👍

File: docs/knative-eventing-mesh/03-01-event-processing.md (hunk 3 above)
Comment (bszwarc) [reply docs/knative-eventing-mesh/03-01-event-processing.md]: we agreed on lambda :)

File: docs/knative-eventing-mesh/03-01-event-processing.md (hunk 8 above)
Comment (bszwarc) [reply docs/knative-eventing-mesh/03-01-event-processing.md]: yes, sounds better. :)

File: docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md [hunk 10]
```diff
@@ -18,32 +17,15 @@ This diagram shows how the Eventing Mesh components work together.
...
-   
-    >**NOTE:** The HTTP Source Adapter accepts only CloudEvents in version 1.0. 
-
-2. The Subscription defines the Broker as the subscriber. This way, the Channel can communicate with the Broker to send events.
-
-3. The Channel listens for incoming events. When it receives an event, the underlying messaging layer dispatches it to the Broker.
+    * Adds the `knative-eventing-injection` label to the user's Namespace. As a result, the Namespace controller creates the [Broker](https://knative.dev/docs/eventing/broker-trigger/) which automatically receives the `default` name. The Broker acts as an entry point for the events which it receives at the cluster-local endpoint it exposes.
 
-4. The Broker sends the event to the Trigger which is configured to receive events of this type. 
+5. The user creates the [Trigger](https://knative.dev/docs/eventing/broker-trigger/) which references the Broker and defines the subscriber along with the conditions for filtering events. This way, certain subscribers receive only the events they are interested in.
```
Comment (kazydek) [inline docs/knative-eventing-mesh/02-01-knative-eventing-mesh.md:29]:
```suggestion
5. The user creates the [Trigger](https://knative.dev/docs/eventing/broker-trigger/) which references the Broker and defines the subscriber along with the conditions for filtering events. This way, subscribers only receive the events of a given type.
```

File: docs/knative-eventing-mesh/03-01-event-processing.md (hunk 4 above)
Comment (kazydek) [reply docs/knative-eventing-mesh/03-01-event-processing.md:9]: now that I look at it, you have all other CRs without "CR" on the diagram so sorry for the confusion :(

**Pull request merged**